### PR TITLE
Added Sonar playbook and step file

### PR DESCRIPTION
**Pull request by samirparhi-dev**
Added Sonar playbook and step file for the sooner scanning. Please review and let me know your comments.

> Good to know
However I did not get the clue on how the print function `__besman_echo_yellow` and `__besman_echo_no_colour` works , I even did not get a clear documentation. It would be great if I can know about those before I use them.

### CODE REVIEW

File: playbooks/besman-sonar-scanner-0.0.1-playbook.sh [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+# Author: Samir Ranjan Parhi
+# License: Same as Repository Licence
+# Usage : Beta-Release
+# Date : 12/02/2024
+
+function __besman_init() {
+    __besman_echo_white "initialising"
+    export ASSESSMENT_TOOL_NAME="sonar-scanner"
+    export ASSESSMENT_TOOL_TYPE="sonar-scanner"
```
Comment (asa1997) [inline playbooks/besman-sonar-scanner-0.0.1-playbook.sh:11]: This should be sast

File: playbooks/besman-sonar-scanner-0.0.1-playbook.sh [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    export ASSESSMENT_TOOL_VERSION="v0.0.1"
```
Comment (asa1997) [inline playbooks/besman-sonar-scanner-0.0.1-playbook.sh:12]: Is the version of sonar-scanner used v0.0.1?

File: playbooks/besman-sonar-scanner-0.0.1-playbook.sh [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    done
+
+    [[ ! -f $BESMAN_TOOL_PATH/$ASSESSMENT_TOOL_NAME ]] && __besman_echo_red "Could not find artifact @ $BESMAN_TOOL_PATH/$ASSESSMENT_TOOL_NAME" && flag=true
+
+    if [[ $flag == true ]]; then
+
+        return 1
+
+    else
+        export SONAR_PATH="$BESMAN_ASSESSMENT_DATASTORE_DIR/$BESMAN_ARTIFACT_NAME/$BESMAN_ARTIFACT_VERSION/sonar"
```
Comment (asa1997) [inline playbooks/besman-sonar-scanner-0.0.1-playbook.sh:56]: "$BESMAN_ASSESSMENT_DATASTORE_DIR/$BESMAN_ARTIFACT_NAME/$BESMAN_ARTIFACT_VERSION/sonar" - > "$BESMAN_ASSESSMENT_DATASTORE_DIR/$BESMAN_ARTIFACT_NAME/$BESMAN_ARTIFACT_VERSION/sast"

File: playbooks/besman-sonar-scanner-0.0.1-playbook.sh [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+
+            __besman_echo_red "Could not find $dir_path"
+
+            flag=true
+
+        fi
+
+    done
+
+    [[ ! -f $BESMAN_TOOL_PATH/$ASSESSMENT_TOOL_NAME ]] && __besman_echo_red "Could not find artifact @ $BESMAN_TOOL_PATH/$ASSESSMENT_TOOL_NAME" && flag=true
```
Comment (asa1997) [inline playbooks/besman-sonar-scanner-0.0.1-playbook.sh:49]: Is this necessary?

File: playbooks/besman-sonar-scanner-0.0.1-playbook.sh [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+function __besman_execute() {
+    local duration
+    __besman_echo_yellow "Launching steps file"
+
+    SECONDS=0
+    . "$BESMAN_STEPS_FILE_PATH"
+    duration=$SECONDS
+
+    export EXECUTION_DURATION=$duration
+    if [[  ==$SONAR_RESULT 1 ]]; then
```
Comment (asa1997) [inline playbooks/besman-sonar-scanner-0.0.1-playbook.sh:76]: Incorrect syntax

File: playbooks/besman-sonar-scanner-0.0.1-steps.sh [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    if ! command -v sonar-scanner &> /dev/null; then
+        echo -e "\n❌ Sonar Scanner CLI is not installed. Please install Sonar Scanner CLI before running this script."
+        SONAR_RESULT=0
+        exit 1
+    fi
+
+    sonar-scanner \
+    -Dsonar.projectKey=$BESMAN_ARTIFACT_NAME \
+    -Dsonar.projectName="$BESMAN_ARTIFACT_NAME" \
+    -Dsonar.host.url=$SONAR_HOST_URL \
```
Comment (asa1997) [inline playbooks/besman-sonar-scanner-0.0.1-steps.sh:37]: Where is this variable set?

**Review by asa1997 — CHANGES_REQUESTED** [review]
@samirparhi-dev Please address the comments.

__besman_echo_yellow "text" is to print the text in yellow color. This function is defined in src/main/bash/besman-utils.sh along with other echo commands.

**Comment by asa1997** [comment]
@samirparhi-dev Please raise this pr from the main branch of your namespace as it would be easier to test.

File: playbooks/besman-sonar-scanner-0.0.1-playbook.sh (hunk 2 above)
Comment (samirparhi-dev) [reply playbooks/besman-sonar-scanner-0.0.1-playbook.sh:12]: is this verison of the tool or version of the variation of this code .

File: playbooks/besman-sonar-scanner-0.0.1-playbook.sh (hunk 3 above)
Comment (samirparhi-dev) [reply playbooks/besman-sonar-scanner-0.0.1-playbook.sh:56]: ok

File: playbooks/besman-sonar-scanner-0.0.1-steps.sh (hunk 6 above)
Comment (samirparhi-dev) [reply playbooks/besman-sonar-scanner-0.0.1-steps.sh:37]: @asa1997 : variable is defined in the playbook init section

File: playbooks/besman-sonar-scanner-0.0.1-playbook.sh (hunk 2 above)
Comment (samirparhi-dev) [reply playbooks/besman-sonar-scanner-0.0.1-playbook.sh:12]:
```
local var_array=("BESMAN_ARTIFACT_TYPE" "BESMAN_ARTIFACT_NAME" "BESMAN_ARTIFACT_VERSION" "BESMAN_ARTIFACT_URL" "BESMAN_ENV_NAME" "BESMAN_ARTIFACT_DIR" "ASSESSMENT_TOOL_NAME" "ASSESSMENT_TOOL_TYPE" "ASSESSMENT_TOOL_VERSION" "ASSESSMENT_TOOL_PLAYBOOK" "BESMAN_ASSESSMENT_DATASTORE_DIR" "BESMAN_TOOL_PATH" "BESMAN_ASSESSMENT_DATASTORE_URL" "BESMAN_LAB_OWNER_TYPE" "BESMAN_LAB_OWNER_NAME" "SONAR_HOST_URL" "SONAR_LOGIN_TOKEN")
```
is this the place we should keep it

File: playbooks/besman-sonar-scanner-0.0.1-steps.sh (hunk 6 above)
Comment (samirparhi-dev) [reply playbooks/besman-sonar-scanner-0.0.1-steps.sh:37]:
```
local var_array=("BESMAN_ARTIFACT_TYPE" "BESMAN_ARTIFACT_NAME" "BESMAN_ARTIFACT_VERSION" "BESMAN_ARTIFACT_URL" "BESMAN_ENV_NAME" "BESMAN_ARTIFACT_DIR" "ASSESSMENT_TOOL_NAME" "ASSESSMENT_TOOL_TYPE" "ASSESSMENT_TOOL_VERSION" "ASSESSMENT_TOOL_PLAYBOOK" "BESMAN_ASSESSMENT_DATASTORE_DIR" "BESMAN_TOOL_PATH" "BESMAN_ASSESSMENT_DATASTORE_URL" "BESMAN_LAB_OWNER_TYPE" "BESMAN_LAB_OWNER_NAME" "SONAR_HOST_URL" "SONAR_LOGIN_TOKEN")
```
is this the right place

**Pull request closed without merging**